### DISCUSSION TITLE
Continuous Injection of particles

### DISCUSSION THREAD

**Discussion by bokkaebi** [BLAST-WarpX/warpx#5180] [Simulation Inputs]
I have two species, electron and ion. I specify some parameters below, but looking at the diagnostics there are no new particles injected other than initially. What parameter should i be changing to inject new particles continuously throughout the simulation?

electron.injection_style            = NRandomPerCell
electron.num_particles_per_cell     = 1e5
electron.momentum_distribution_type = "maxwell_boltzmann"
electron.theta_distribution_type    = constant
electron.theta                      = k_b*(temperature)/(m_e)/c/c
electron.do_continuous_injection    = 1
electron.profile                    = "constant"
electron.density                    = 1e21

**Comment by n01r** [comment]
Hi @bokkaebi,

Thank you for your question! :)
It seems to me that instead of the injector `NRandomPerCell`, the injector `NFluxPerCell` may be more suited to what you have in mind. Please check out this answer in one of our issues and let me know if this has what you are looking for.
https://github.com/ECP-WarpX/WarpX/issues/5131#issuecomment-2287657798

The parameter `do_continuous_injection` for other injection methods only controls whether or not additional particles will be injected if the bounding box moves in space. That is usually the case for a moving window simulation or one done in the boosted frame.
